sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -200,31 +203,21 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
    * @param locationId location id
    * @returns InstitutionLocation
    */
-  async getLocationById(locationId: number): Promise<InstitutionLocation> {
-    return this.repo
-      .createQueryBuilder("location")
-      .select(["location.name", "location.id"])
-      .where("location.id = :locationId", { locationId })
-      .getOne();
-  }
-
-  /**
-   * Get institution location by location id.
-   * @param locationId location id
-   * @returns InstitutionLocation
-   */
-  async getDesignatedLocationById(
+  async getLocation(
     locationId: number,
-  ): Promise<InstitutionLocation> {
+  ): Promise<DesignatedAndNotDesignatedLocations> {
     return this.repo
       .createQueryBuilder("location")
-      .select(["location.name"])
-      .where("location.id = :locationId", { locationId })
-      .andWhere(
-        `EXISTS(${this.designationAgreementLocationService
-          .getExistsDesignatedLocation()
-          .getSql()})`,
+      .select("location.name", "locationName")
+      .addSelect(
+        `CASE
+          WHEN EXISTS(${this.designationAgreementLocationService
+            .getExistsDesignatedLocation()
+            .getSql()}) THEN true
+          ELSE false
+        END "isDesignated"`,
       )
-      .getOne();
+      .where("location.id = :locationId", { locationId })
+      .getRawOne();
```

Comment:
We could add a transient property to the entity(isDesignated) and use getOne() instead of getRawOne(). 
If the query results deviate a lot from Entity model, then service model will be more optimal. may be in this case transient property can help IMO.

Reply:
i didn't understand? could you explain